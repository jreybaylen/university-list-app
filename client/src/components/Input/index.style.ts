import { StypesProps } from '@interface/style.interface'

const styles: StypesProps = {
    input: {
        outline: 0,
        width: '100%',
        borderRadius: 4,
        padding: '10px 15px',
        border: '1px solid #ccc',
        textTransform: 'capitalize'
    }
}

export {
    styles
}