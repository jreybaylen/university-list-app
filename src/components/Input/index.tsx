import { ChangeEvent, InputHTMLAttributes } from 'react'

import { styles } from './index.style'

function Input (props: InputHTMLAttributes<HTMLInputElement>): JSX.Element {
    const { onChange, ...rest } = props
    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
        if (props.onChange) {
            props.onChange(event)
        }
    }
    const inputElement = (
        <input style={ styles.input } onChange={ handleChange } { ...rest } />
    )

    return inputElement
}

export {
    Input
}