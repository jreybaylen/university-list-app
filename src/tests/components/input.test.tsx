import { render, fireEvent } from '@testing-library/react'

import { Input } from '@components/Input'

describe('<Input /> Component', () => {
    it('Should render without crashing', () => {
        const wrapper = render(<Input />)

        expect(wrapper).toMatchSnapshot()
    })
    
    it('Should render input when using the component', () => {
        const { container } = render(<Input />)

        expect(container.querySelector('input')).toBeInTheDocument()
    })

    it('Should have dynamic properties', () => {
        const padding = '8px 10px'
        const placeholder = "Testing"
        const { getByDisplayValue } = render(
            <Input
                value={ 12 }
                style={{ padding }}
                placeholder={ placeholder }
            />
        )
        const inputElement = getByDisplayValue(12)

        expect(inputElement).toBeInTheDocument()
        expect(inputElement).toHaveDisplayValue('12')
        expect(inputElement).toHaveStyle(`padding: ${ padding }`)
        expect(inputElement.getAttribute('placeholder')).toBe(placeholder)
    })

    it('Should trigger onChange when the component detected the value has been changed', () => {
        const testId = 'input-element'
        const handleInputChange = jest.fn()
        const { getByLabelText } = render(
            <Input onChange={ handleInputChange } aria-label={ testId } />
        )
        const inputElement = getByLabelText(testId)

        expect(inputElement).toHaveDisplayValue('')
        fireEvent.change(inputElement, { target: { value: 26 } })
        expect(handleInputChange).toHaveBeenCalled()
        expect(inputElement).toHaveDisplayValue('26')
    })
})