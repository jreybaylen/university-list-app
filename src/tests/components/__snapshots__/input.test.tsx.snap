// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Component Should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        style="outline: 0; width: 100%; font-weight: normal; padding: 10px 15px; border: 1px solid #ccc;"
      />
    </div>
  </body>,
  "container": <div>
    <input
      style="outline: 0; width: 100%; font-weight: normal; padding: 10px 15px; border: 1px solid #ccc;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
